k8s-intf/src/bolero/support.rs

```diff
@@ -0,0 +1,584 @@
+    d: &mut D,
+    chars: &str,
+    min: std::ops::Bound<&usize>,
+    max: std::ops::Bound<&usize>,
+) -> Option<String> {
+    let len = d.gen_usize(min, max)?;
+    (0..len)
+        .map(|_| {
+            chars
+                .chars()
+                .nth(d.gen_usize(Bound::Included(&0), Bound::Excluded(&chars.len()))?)
+        })
+        .collect()
+}
+
+pub struct Ipv4AddrString(pub String);
+
+impl TypeGenerator for Ipv4AddrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        Some(Ipv4AddrString(
+            Ipv4Addr::from(d.gen_u32(
+                Bound::Included(&0x1000_0000_u32),
+                Bound::Excluded(&0xffff_ffff_u32),
+            )?)
+            .to_string(),
+        ))
+    }
+}
+
+pub struct Ipv6AddrString(pub String);
+
+impl TypeGenerator for Ipv6AddrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        Some(Ipv6AddrString(
+            Ipv6Addr::from(d.gen_u128(
+                Bound::Included(&0x0000_0000_0000_0000_0000_0000_0000_0001_u128),
+                Bound::Excluded(&0xffff_ffff_ffff_ffff_ffff_ffff_ffff_ffff_u128),
+            )?)
+            .to_string(),
+        ))
+    }
+}
+
+pub struct IpAddrString(pub String);
+
+impl TypeGenerator for IpAddrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        let is_ipv4 = d.gen_bool(None)?;
+        if is_ipv4 {
+            Some(IpAddrString(d.produce::<Ipv4AddrString>()?.0))
+        } else {
+            Some(IpAddrString(d.produce::<Ipv6AddrString>()?.0))
+        }
+    }
+}
+
+pub struct V4CidrString(pub String);
+pub struct V6CidrString(pub String);
+pub struct CidrString(pub String);
+
+fn v4cdir_from_bytes(addr_bytes: u32, mask: u8) -> String {
+    // Remove this allow once we upgrade to Rust 1.87.0
+    #[allow(unstable_name_collisions)]
+    let and_mask = u32::MAX.unbounded_shl(32 - u32::from(mask));
+    let addr = Ipv4Addr::from(addr_bytes & and_mask);
+    format!("{addr}/{mask}")
+}
+
+fn v6cdir_from_bytes(addr_bytes: u128, mask: u8) -> String {
+    // Remove this allow once we upgrade to Rust 1.87.0
+    #[allow(unstable_name_collisions)]
+    let and_mask = u128::MAX.unbounded_shl(128 - u32::from(mask));
+    let addr = Ipv6Addr::from(addr_bytes & and_mask);
+    format!("{addr}/{mask}")
+}
+
+impl TypeGenerator for V4CidrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        let mask = d.gen_u8(Bound::Included(&0), Bound::Included(&32))?;
+        let addr_bytes = d.produce::<u32>()?;
+        Some(V4CidrString(v4cdir_from_bytes(addr_bytes, mask)))
+    }
+}
+
+impl TypeGenerator for V6CidrString {
+    fn generate<D: Driver>(d: &mut D) -> Option<Self> {
+        let mask: u8 = d.gen_u8(Bound::Included(&0), Bound::Included(&128))?;
+        let addr_bytes = d.produce::<u128>()?;
+        Some(V6CidrString(v6cdir_from_bytes(addr_bytes, mask)))
+    }
+}
+
+#[derive(Debug)]
+pub struct UniqueV4CidrGenerator {
+    count: u16,
+    mask: u8,
```

Comment:
I seem to recall writing a (potentially) helpful type for Ip{4,6,}Mask which bounded the range of the u8.  That might make these type generators less tricky, but it may be overkill as well

Reply:
Yep, that would be better but not worth the time, IMO